src/Pos/Types/Block/Types.hs

```diff
@@ -58,7 +58,12 @@ import           Pos.Ssc.Class.Types (Ssc (..))
 data MainBlockchain ssc
 
 -- | Data to be signed in main block.
-type MainToSign ssc = (HeaderHash, BodyProof (MainBlockchain ssc), SlotId, ChainDifficulty)
+type MainToSign ssc = ( HeaderHash
```

Comment:
Let's wrap it into `data`. Tuple with 5 elements makes people sad, especially @ChShersh.

Also we need to think about ensuring that everything is here, so that if later someone adds something to block or header it is guaranteed to be signed.